src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1017,6 +1024,46 @@ private function registerPropertyInfoConfiguration(array $config, ContainerBuild
         }
     }
 
+    private function registerCacheConfiguration(array $config, ContainerBuilder $container)
+    {
+        foreach ($config['adapters'] as $name => $adapter) {
+            $class = null;
+            $arguments = array();
+            $namespaceArgumentIndex = null;
+
+            switch ($adapter['type']) {
+                case 'apcu':
+                    $class = ApcuAdapter::class;
+                    $arguments[] = null;
+                    $arguments[] = isset($adapter['options']['default_lifetime']) ? $adapter['options']['default_lifetime'] : 0;
+                    $namespaceArgumentIndex = 0;
+                    break;
+                case 'doctrine':
+                    $class = DoctrineAdapter::class;
+                    $arguments[] = isset($adapter['options']['cache_provider_service']) ? new Reference($adapter['options']['cache_provider_service']) : null;
+                    $arguments[] = isset($adapter['options']['default_lifetime']) ? $adapter['options']['default_lifetime'] : null;
+                    break;
+                case 'filesystem':
+                    $class = FilesystemAdapter::class;
+                    $arguments[] = isset($adapter['options']['directory']) ? $adapter['options']['directory'] : null;
+                    $arguments[] = isset($adapter['options']['default_lifetime']) ? $adapter['options']['default_lifetime'] : null;
+                    break;
```

Comment:
I don't think we need that here as we already validate the type in the bundle configuration.